packages/flutter_tools/tool/tool_coverage.dart

```diff
@@ -76,11 +76,15 @@ class VMPlatform extends PlatformPlugin {
     final ServiceProtocolInfo info = await Service.controlWebServer(enable: true);
     final dynamic channel = IsolateChannel<Object>.connectReceive(receivePort)
         .transformStream(StreamTransformer<Object, Object>.fromHandlers(handleDone: (EventSink<Object> sink) async {
-      await coverageCollector.collectCoverageIsolate(info.serverUri);
-      isolate.kill(priority: Isolate.immediate);
-      isolate = null;
-      sink.close();
-      completer.complete();
+      try {
+        // this will throw if collection fails.
+        await coverageCollector.collectCoverageIsolate(info.serverUri);
+      } finally {
+        isolate.kill(priority: Isolate.immediate);
```

Comment:
Hopefully the IsolateChannel knows at this point that it owns the receivePort.

Reply:
I think by construction this will always be true , but if it isn't then at worst we'll leak some isolates during the test process